UhooiPicBook/Modules/MonsterDetail/Views/MonsterDetailViewController.swift

```diff
@@ -73,11 +77,11 @@ final class MonsterDetailViewController: UIViewController {
     }
 
     private func configureView() {
-        self.imageCacheManager.cacheImage(imageUrl: monster.iconUrl) { result in
+        self.imageCacheManager.cacheImage(imageUrl: monster.iconUrl) { [weak self] result in
             switch result {
             case let .success(icon):
                 DispatchQueue.main.async {
```

Comment:
ここもアプリの foreground/background 状態次第ではメモリリークすることがありえますね

```suggestion
                DispatchQueue.main.async { [weak self] in
```

Reply:
うおおありがとうございます :bow:
メインスレッドで実行しているのでメモリリークはあり得ないと思っていましたが、あり得るのですね…。

Reply:
ここですが、weak self は基本内側のクロージャに伝播するので外側で weak self していれば問題ないです　例外として外側でguard let self = self してる時などは必要です